polaris-react/src/components/Box/Box.tsx

```diff
@@ -327,10 +335,12 @@ export const Box = forwardRef<HTMLElement, BoxProps>(
         : undefined,
       zIndex,
       opacity,
+      ...getComponentProperties(layout?.customProperties),
```

Comment:
Strips out any custom styles, and only allows custom properties starting with '--pc'